File: Java_Assignment/Assignment_49/programAQ3.java
///////////////////////////////////////////////////////////
//
//  File name : Assignement49Question3.java
//  Description : Write Java program which accept String from user and display
//                 below pattern.
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//  
///////////////////////////////////////////////////////////

/*
Input : Hello
Output : 
H # # # #
H e # # #
H e l # #
H e l l #
H e l l o

Program Layout :
*/

import java.util.Scanner;

class Pattern
{
    public void PatternX(String str)
    {
        int i = 0,j = 0;

        for(i = 1;i < str.length();i++)
        {        
            for(j = 0;j < str.length();j++)
            {
                if(j < i)
                {
                    System.out.print(str.charAt(j) + "\t");
                }
                else
                {
                    System.out.print("#\t");
                }    
            }
            System.out.println();
        }
    }
}

public class programAQ3
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);
        String str = "";

        System.out.println("Enter String: ");
        str = sobj.nextLine();

        Pattern pobj = new Pattern();

        pobj.PatternX(str);

        sobj.close();
    }   
}

File: C_Assignment/Assignment_32/programAQ3.c
///////////////////////////////////////////////////////////
//
//  File name : Assignement32Question3.c
//  Description : Write a program which accept one number and position from user and on
//                that bit. Return modified number.
//  Author : Kartik Ganesh Jare
//  Date : 13/11/25
//  
///////////////////////////////////////////////////////////

/*

Input : 10 3
Output : 14

*/

#include<stdio.h>

typedef unsigned int UINT;

UINT ONBit(UINT iNo,UINT iPos)
{
    UINT iMask = 1;
    UINT iResult = 0;

    iMask =  iMask << (iPos - 1);

    iResult = iNo | iMask;

    return iResult;
}

int main()
{
    UINT iValue = 0,iLocation = 0,iRet = 0;

    printf("Enter number: \n");
    scanf("%u",&iValue);

    printf("Enter the postion : \n");
    scanf("%u",&iLocation);

    iRet = ONBit(iValue,iLocation);

    printf("changed number is : %u\n",iRet);

    return 0;
}